File: dofbot_moveit/scripts/.ipynb_checkpoints/02_motion_plan-checkpoint.py
#!/usr/bin/env python
# coding: utf-8
import rospy
from math import pi
from time import sleep
from geometry_msgs.msg import Pose
from moveit_commander.move_group import MoveGroupCommander
from tf.transformations import quaternion_from_euler


DE2RA = pi / 180

if __name__ == '__main__':

    rospy.init_node("dofbot_motion_plan_py")

    dofbot = MoveGroupCommander("dofbot")

    dofbot.set_named_target("down")
    dofbot.go()
    sleep(0.5)

    dofbot.allow_replanning(True)
    dofbot.set_planning_time(5)

    dofbot.set_num_planning_attempts(10)

    dofbot.set_goal_position_tolerance(0.01)

    dofbot.set_goal_orientation_tolerance(0.01)

    dofbot.set_goal_tolerance(0.01)

    dofbot.set_max_velocity_scaling_factor(1.0)

    dofbot.set_max_acceleration_scaling_factor(1.0)

    pos = Pose()

    pos.position.x = 0.0
    pos.position.y = 0.0597016
    pos.position.z = 0.168051

    roll = -140.0
    pitch = 0.0
    yaw = 0.0

    q = quaternion_from_euler(roll * DE2RA, pitch * DE2RA, yaw * DE2RA)
    # pos.orientation.x = 0.940132
    # pos.orientation.y = -0.000217502
    # pos.orientation.z = 0.000375234
    # pos.orientation.w = -0.340811
    pos.orientation.x = q[0]
    pos.orientation.y = q[1]
    pos.orientation.z = q[2]
    pos.orientation.w = q[3]

    dofbot.set_pose_target(pos)

    for i in range(5):

        plan = dofbot.plan()
        if len(plan.joint_trajectory.points) != 0:
            print ("plan success")

            dofbot.execute(plan)
            break
        else:
            print ("plan error")

File: dofbot_moveit/src/.ipynb_checkpoints/02_motion_plan-checkpoint.cpp
#include <iostream>
#include "ros/ros.h"
#include <moveit/move_group_interface/move_group_interface.h>
#include <tf/LinearMath/Quaternion.h>

using namespace std;

const float DE2RA = M_PI / 180.0f;

int main(int argc, char **argv) {

    ros::init(argc, argv, "dofbot_motion_plan_cpp");

    ros::NodeHandle n;

    ros::AsyncSpinner spinner(1);
    spinner.start();

    moveit::planning_interface::MoveGroupInterface dofbot("dofbot");

    dofbot.allowReplanning(true);

    dofbot.setPlanningTime(5);

    dofbot.setNumPlanningAttempts(10);

    dofbot.setGoalPositionTolerance(0.01);

    dofbot.setGoalOrientationTolerance(0.01);

    dofbot.setMaxVelocityScalingFactor(1.0);

    dofbot.setMaxAccelerationScalingFactor(1.0);

    geometry_msgs::Pose pose;
    pose.position.x = 0.0;
    pose.position.y = 0.05957241;
    pose.position.z = 0.1680498;

    tf::Quaternion quaternion;

    double Roll = -140;
    double Pitch = 0.0;
    double Yaw = 0.0;

    quaternion.setRPY(Roll * DE2RA, Pitch * DE2RA, Yaw * DE2RA);
//    pose.orientation.x = 0.940132;
//    pose.orientation.y = -0.000217502;
//    pose.orientation.z = 0.000375234;
//    pose.orientation.w = -0.340811;
    pose.orientation.x = quaternion.x();
    pose.orientation.y = quaternion.y();
    pose.orientation.z = quaternion.z();
    pose.orientation.w = quaternion.w();
    string link = dofbot.getEndEffectorLink();

    dofbot.setPoseTarget(pose, link);
    int index = 0;

    while (index <= 10) {
        moveit::planning_interface::MoveGroupInterface::Plan plan;

        const moveit::planning_interface::MoveItErrorCode &code = dofbot.plan(plan);
        if (code == code.SUCCESS) {
            ROS_INFO_STREAM("plan success");
            dofbot.execute(plan);
            break;
        } else {
            ROS_INFO_STREAM("plan error");
        }
        index++;
    }
    return 0;
}

